File: frontend/src/api/modules/review.ts
import request from '../request'

export interface ReviewTaskDTO {
  id: number
  repo_id: number
  repo_key: string
  repo_name: string
  provider_config_id: number
  platform: string
  event_type: string
  mr_iid: string
  source_branch: string
  target_branch: string
  commit_sha: string
  trigger_type: string
  trigger_user: string
  status: string
  risk_level: string
  summary: string
  error_message: string
  started_at: string | null
  finished_at: string | null
  created_at: string
  updated_at: string
  findings_count: number
  raw_diff: string
}

export interface ReviewFindingDTO {
  id: number
  task_id: number
  source: string
  rule_id: string
  severity: string
  file_path: string
  old_line: number
  new_line: number
  title: string
  message: string
  suggestion: string
  fingerprint: string
}

export interface MergeCheckDTO {
  mergeable: boolean
  checks: MergeCheckItemDTO[]
}

export interface MergeCheckItemDTO {
  check_type: string
  status: string
  risk_level: string
  message: string
}

export function createReviewTask(data: {
  repo_key: string
  provider_config_id?: number
  mr_iid: string
  commit_sha?: string
  trigger_type?: string
}) {
  return request.post<unknown, ReviewTaskDTO>('/reviews/tasks', data)
}

export function getReviewTask(id: number) {
  return request.get<unknown, ReviewTaskDTO>(`/reviews/tasks/${id}`)
}

export function listReviewTasks(params: {
  repo_key: string
  status?: string
  page?: number
  page_size?: number
}) {
  return request.get<unknown, {
    tasks: ReviewTaskDTO[]
    pagination: { total: number; page: number; page_size: number }
  }>('/reviews/tasks', { params })
}

export function listReviewFindings(task_id: number, params?: { severity?: string; source?: string }) {
  return request.get<unknown, ReviewFindingDTO[]>(`/reviews/tasks/${task_id}/findings`, { params })
}

export function retryReviewTask(id: number, data?: { owner?: string; repo?: string }) {
  return request.post<unknown, ReviewTaskDTO>(`/reviews/tasks/${id}/retry`, data || {})
}

export function checkMerge(params: { repo_key: string; mr_iid: string; commit_sha?: string }) {
  return request.get<unknown, MergeCheckDTO>('/merge-checks', { params })
}

export function getReviewConfig(repo_key: string) {
  return request.get<unknown, { config_yaml: string }>(`/reviews/config/${repo_key}`)
}

export function updateReviewConfig(repo_key: string, configYaml: string) {
  return request.put<unknown, { config_yaml: string }>(`/reviews/config/${repo_key}`, { config_yaml: configYaml })
}

export interface ReviewRepoConfigDTO {
  id: number
  provider_config_id: number
  platform_owner: string
  platform_repo: string
  enabled: boolean
  block_on_high: boolean
  auto_review_on_mr: boolean
  llm_provider: string
  max_files: number
  max_diff_lines: number
  rule_overrides_json: string
  scope_note: string
  linked_repos: LinkedRepoDTO[]
  prompt_prefix_override: string
  prompt_intent_override: string
}

export interface LinkedRepoDTO {
  id: number
  key: string
  name: string
}

export function getRemoteRepoConfig(provider_id: number, owner: string, repo: string) {
  return request.get<unknown, ReviewRepoConfigDTO>(`/review/remote-config/${provider_id}/${owner}/${repo}`)
}

export function updateRemoteRepoConfig(provider_id: number, owner: string, repo: string, data: Partial<ReviewRepoConfigDTO>) {
  return request.put<unknown, ReviewRepoConfigDTO>(`/review/remote-config/${provider_id}/${owner}/${repo}`, data)
}

export function createReviewTaskByProvider(data: {
  provider_config_id: number
  owner: string
  repo: string
  mr_iid: string
  commit_sha?: string
  trigger_type?: string
}) {
  return request.post<unknown, ReviewTaskDTO>('/reviews/tasks/provider', data)
}

export function listReviewTasksByProvider(params: {
  provider_id: number
  mr_iid: string
  page?: number
  page_size?: number
}) {
  return request.get<unknown, {
    tasks: ReviewTaskDTO[]
    pagination: { total: number; page: number; page_size: number }
  }>('/reviews/tasks/provider', { params })
}

export interface ReviewStatsDTO {
  total_reviews: number
  total_findings: number
  pass_count: number
  blocked_count: number
  failed_count: number
  pass_rate: number
  by_risk_level: Record<string, number>
  by_severity: Record<string, number>
  by_source: Record<string, number>
  by_rule: Array<{ name: string; count: number }>
  top_files: Array<{ name: string; count: number }>
  daily_trend: Array<{
    date: string
    total: number
    passed: number
    rate: number
  }>
}

export function getReviewStats(params?: { repo_key?: string; period?: string }) {
  return request.get<unknown, ReviewStatsDTO>('/reviews/stats', { params })
}

export function submitFindingFeedback(findingId: number, feedback: 'useful' | 'false_positive') {
  return request.post<unknown, { status: string }>(`/reviews/findings/${findingId}/feedback`, { feedback })
}

export interface RAGIndexResult {
  repo_key: string
  chunk_count: number
  file_count: number
  duration: number
  error: string
}

export interface RAGStatsResult {
  stats: Record<string, number>
  available: boolean
}

export function indexRepoRAG(repo_key: string) {
  return request.post<unknown, RAGIndexResult>(`/reviews/rag/index/${repo_key}`)
}

export function getRAGStats() {
  return request.get<unknown, RAGStatsResult>('/reviews/rag/stats')
}

File: biz/model/api/review.go
package api

import (
	"time"

	"github.com/yi-nology/git-manage-service/biz/model/po"
)

type ReviewTaskDTO struct {
	ID               uint       `json:"id"`
	RepoID           uint       `json:"repo_id"`
	RepoKey          string     `json:"repo_key"`
	RepoName         string     `json:"repo_name"`
	ProviderConfigID uint       `json:"provider_config_id"`
	Platform         string     `json:"platform"`
	EventType        string     `json:"event_type"`
	MRIID            string     `json:"mr_iid"`
	SourceBranch     string     `json:"source_branch"`
	TargetBranch     string     `json:"target_branch"`
	CommitSHA        string     `json:"commit_sha"`
	TriggerType      string     `json:"trigger_type"`
	TriggerUser      string     `json:"trigger_user"`
	Status           string     `json:"status"`
	RiskLevel        string     `json:"risk_level"`
	Summary          string     `json:"summary"`
	ErrorMessage     string     `json:"error_message"`
	StartedAt        *time.Time `json:"started_at"`
	FinishedAt       *time.Time `json:"finished_at"`
	CreatedAt        time.Time  `json:"created_at"`
	UpdatedAt        time.Time  `json:"updated_at"`
	FindingsCount    int        `json:"findings_count"`
	RawDiff          string     `json:"raw_diff"`
}

func NewReviewTaskDTO(t po.ReviewTask) ReviewTaskDTO {
	return ReviewTaskDTO{
		ID:               t.ID,
		RepoID:           t.RepoID,
		ProviderConfigID: t.ProviderConfigID,
		Platform:         t.Platform,
		EventType:        t.EventType,
		MRIID:            t.MRIID,
		SourceBranch:     t.SourceBranch,
		TargetBranch:     t.TargetBranch,
		CommitSHA:        t.CommitSHA,
		TriggerType:      t.TriggerType,
		TriggerUser:      t.TriggerUser,
		Status:           t.Status,
		RiskLevel:        t.RiskLevel,
		Summary:          t.Summary,
		ErrorMessage:     t.ErrorMessage,
		RawDiff:          t.RawDiff,
		StartedAt:        t.StartedAt,
		FinishedAt:       t.FinishedAt,
		CreatedAt:        t.CreatedAt,
		UpdatedAt:        t.UpdatedAt,
	}
}

type ReviewFindingDTO struct {
	ID          uint   `json:"id"`
	TaskID      uint   `json:"task_id"`
	Source      string `json:"source"`
	RuleID      string `json:"rule_id"`
	Severity    string `json:"severity"`
	FilePath    string `json:"file_path"`
	OldLine     int    `json:"old_line"`
	NewLine     int    `json:"new_line"`
	Title       string `json:"title"`
	Message     string `json:"message"`
	Suggestion  string `json:"suggestion"`
	Fingerprint string `json:"fingerprint"`
}

func NewReviewFindingDTO(f po.ReviewFinding) ReviewFindingDTO {
	return ReviewFindingDTO{
		ID:          f.ID,
		TaskID:      f.TaskID,
		Source:      f.Source,
		RuleID:      f.RuleID,
		Severity:    f.Severity,
		FilePath:    f.FilePath,
		OldLine:     f.OldLine,
		NewLine:     f.NewLine,
		Title:       f.Title,
		Message:     f.Message,
		Suggestion:  f.Suggestion,
		Fingerprint: f.Fingerprint,
	}
}

type ReviewRepoConfigDTO struct {
	ID                   uint            `json:"id"`
	ProviderConfigID     uint            `json:"provider_config_id"`
	PlatformOwner        string          `json:"platform_owner"`
	PlatformRepo         string          `json:"platform_repo"`
	Enabled              bool            `json:"enabled"`
	BlockOnHigh          bool            `json:"block_on_high"`
	AutoReviewOnMR       bool            `json:"auto_review_on_mr"`
	LLMProvider          string          `json:"llm_provider"`
	MaxFiles             int             `json:"max_files"`
	MaxDiffLines         int             `json:"max_diff_lines"`
	RuleOverrides        string          `json:"rule_overrides_json"`
	ScopeNote            string          `json:"scope_note"`
	LinkedRepos          []LinkedRepoDTO `json:"linked_repos"`
	PromptPrefixOverride string          `json:"prompt_prefix_override"`
	PromptIntentOverride string          `json:"prompt_intent_override"`
}

type LinkedRepoDTO struct {
	ID   uint   `json:"id"`
	Key  string `json:"key"`
	Name string `json:"name"`
}

func NewReviewRepoConfigDTO(c po.ReviewRepoConfig, repos []LinkedRepoDTO) ReviewRepoConfigDTO {
	return ReviewRepoConfigDTO{
		ID:                   c.ID,
		ProviderConfigID:     c.ProviderConfigID,
		PlatformOwner:        c.PlatformOwner,
		PlatformRepo:         c.PlatformRepo,
		Enabled:              c.Enabled,
		BlockOnHigh:          c.BlockOnHigh,
		AutoReviewOnMR:       c.AutoReviewOnMR,
		LLMProvider:          c.LLMProvider,
		MaxFiles:             c.MaxFiles,
		MaxDiffLines:         c.MaxDiffLines,
		RuleOverrides:        c.RuleOverridesJSON,
		ScopeNote:            c.ScopeNote,
		LinkedRepos:          repos,
		PromptPrefixOverride: c.PromptPrefixOverride,
		PromptIntentOverride: c.PromptIntentOverride,
	}
}

type MergeCheckDTO struct {
	Mergeable bool                `json:"mergeable"`
	Checks    []MergeCheckItemDTO `json:"checks"`
}

type MergeCheckItemDTO struct {
	CheckType string `json:"check_type"`
	Status    string `json:"status"`
	RiskLevel string `json:"risk_level"`
	Message   string `json:"message"`
}
